modules/storage-ephemeral-vols-overview.adoc

```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * storage/generic-ephemeral-vols.adoc
+
+:_content-type: CONCEPT
+[id="generic-ephemeral-vols-overview_{context}"]
+= Overview
+
+Generic ephemeral volumes are a type of ephemeral volume that can be provided by all storage drivers that also support persistent volumes and dynamic provisioning. Generic ephemeral volumes are similar to `emptyDir` volumes in that they provide a per-pod directory for scratch data, which is usually empty after provisioning. Generic ephemeral volumes are specified inline in the pod spec.
+
+Generic ephemeral volumes have the following features:
+
+* Storage can be local or network-attached.
+
+* Volumes can have a fixed size that pods are not able to exceed.
+
+* Volumes might have some initial data, depending on the driver and parameters.
+
+* Typical operations on volumes are supported, assuming that the driver supports them, including snapshotting, cloning, resizing, and storage capacity tracking.
```

Comment:
Hi @lpettyjo , Seems there's some limit about the driver support features like snapshotting, cloning, resizing. As the Generic ephemeral volumes has the same lifecycle with pod level if the driver only support offline(means snapshotting, cloning, resizing need the volume detach from the node then do the snapshotting, cloning, resizing action) features, it maybe not work with Generic ephemeral volumes.
E.g. Azure-disk CSI driver only support offline resize, need volume detach from the node and do the resize action but acually if we detach volume from the node means pod needs recreate also the related Generic ephemeral volumes will also recreate then the data will be lost, but as the support resize feature means after resize the origin data will still exist.

Reply:
@lpettyjo Good point. AFAIK, only Azure Disk CSI driver only supports offline resize, vSphere CSI Driver on vCenter version 6.7.3~7.0.1 only support offline resize, from 7.0.2+ support online resize [doc](https://cormachogan.com/2021/04/16/vsphere-csi-v2-2-online-volume-expansion/) . In addition, cinder CSI only support offline snapshot but generic ephemeral volume need volume always online, should not support. Thanks for point out add enchancement for azure.

Reply:
@Phaow Thanks for info. I believe then that I should indicate explicitly that Azure Disk and Cinder do not support resize. I think that I don't need to make this statement for vSphere because we have removed support for 6.7U2 and earlier and recommend 7.0U2 or later. Sound right to you?

Reply:
Hmm, seems need a bit modify. Azure Disk CSI do not support resize, Cinder CSI do not support snapshot for "Generic ephemeral volumes".

Reply:
Thanks for the refinement. I've pushed this change.